pyproject.toml

```diff
@@ -27,6 +27,7 @@ dependencies = [
     "spurplus ~= 2.3.4",
     "websockets ~= 10.3",
     "charset_normalizer ~= 2.1.1",
+    "msrestazure ~= 0.6.4",
```

Comment:
Don't use it in default dependencies. As comments in another PR. It needs to create another installation option or put it in Azure staffs.

Reply:
I moved it to the default dependencies, because in CI check, the example parts will fail because lack of msrest package. so I need update noxfile.py in example method to use the split installation option?

Reply:
Here is the call trace.
```
nox > lisa --debug
Traceback (most recent call last):
  File "/home/runner/work/lisa/lisa/.nox/example/bin/lisa", line 5, in <module>
    from lisa.main import cli
  File "/home/runner/work/lisa/lisa/lisa/__init__.py", line 6, in <module>
    from lisa.environment import Environment
  File "/home/runner/work/lisa/lisa/lisa/environment.py", line 19, in <module>
    from lisa.node import Node, Nodes
  File "/home/runner/work/lisa/lisa/lisa/node.py", line 24, in <module>
    from lisa.nic import Nics, NicsBSD
  File "/home/runner/work/lisa/lisa/lisa/nic.py", line 15, in <module>
    from lisa.tools import Cat, Ip, KernelConfig, Ls, Lspci, Modprobe, Tee
  File "/home/runner/work/lisa/lisa/lisa/tools/__init__.py", line 93, in <module>
    from .reboot import Reboot
  File "/home/runner/work/lisa/lisa/lisa/tools/reboot.py", line 13, in <module>
    from lisa.features import SerialConsole
  File "/home/runner/work/lisa/lisa/lisa/features/__init__.py", line 14, in <module>
    from .gpu import Gpu, GpuEnabled, GpuSettings
  File "/home/runner/work/lisa/lisa/lisa/features/gpu.py", line 17, in <module>
    from lisa.sut_orchestrator.azure.tools import LisDriver
  File "/home/runner/work/lisa/lisa/lisa/sut_orchestrator/azure/tools.py", line 9, in <module>
    from msrestazure.azure_cloud import AZURE_PUBLIC_CLOUD, Cloud  # type: ignore
ModuleNotFoundError: No module named 'msrestazure'
```

Reply:
The general gpu feature referred an Azure tool, it's not right. If the code related to Azure, it shouldn't be in a general feature. Please move the logic from general gpu feature to azure extended gpu. But is the LisDriver still useful?

Reply:
It needs to move `import lisa.transformers.disable_cloud_components` into `try/except` of Azure modules. The general GPU feature shouldn't use lisdriver. The related logic should be in Azure GPU feature.

Reply:
updated, thanks.